### DISCUSSION TITLE
Conditional step always evaluating to "True"

### DISCUSSION THREAD

**Discussion by jsmith-itscovered** [community/community#57265] [Actions] [answered]
### Select Topic Area

Bug

### Body

I'm trying to set up a workflow for feature branches, where they will deploy a java package from the feature branch, only if the version has 'SNAPSHOT' in it.

Here's my code:

`        name: Get version

        id: version
        
        run: |
        
          echo "TAG=$(./gradlew properties -q | grep \"version:\" | awk '{print $2}')" >> $GITHUB_OUTPUT
          
      -
        name: Publish if SNAPSHOT
        
        if: ${{contains('SNAPSHOT',steps.version.outputs.TAG_NAME)}}
        
        run: |
        
          ./gradlew publish
          `
          
My problem is: the conditional always evaluates to true. Meaning developers can deploy major releases from the feature branch. We want them to be able to deploy snapshot builds ONLY, from feature branch (in our gradle.build file, snapshots and releases are actually two different repositories). 

I can't think of any reason for that to evaluate to true, other than a bug. It's certainly not finding the string 'SNAPSHOT', in the version field, in our gradle.build file.

**Comment by ChristopherHX** [comment, marked as answer]
Can you try to swap the parameters of contains? It looks for `steps.version.outputs.TAG_NAME` in `'SNAPSHOT'`, which is not what you are trying to do.

```yaml
    if: ${{contains(steps.version.outputs.TAG_NAME, 'SNAPSHOT')}}
```

I would guess your condition evaluated to true, because `steps.version.outputs.TAG_NAME` was empty. Maybe consider enabling step debugging (e.g. it is part of rerun jobs UI), then you get a verbose list of the operations inside expressions.

_BTW your output `echo "TAG=$(./gradlew properties -q | grep \"version:\" | awk '{print $2}')" >> $GITHUB_OUTPUT` has name `TAG` and not `TAG_NAME`._

**Comment by jsmith-itscovered** [comment]
I did as you said (sorry about the TAG_NAME, that was correct at some point, but then I decided to change it in only one place), but now the expression always evaluates to false.

**Reply by ChristopherHX** [reply]
No idea due to lack of information. (e.g. how your workflow looks like right now and debug logs)

Enable [enabling-step-debug-logging](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging#enabling-step-debug-logging) and paste the log of the step setting the output and the skipped step.

**Reply by jsmith-itscovered** [reply]
nvm. figured it out. had to use a --no-daemon flag on my gradle properties:

`./gradlew properties --no-daemon --console=plain -q | grep "^version:" | awk '{printf $2}'`
